sender/api/src/main/java/org/phenotips/integration/lims247/internal/RemoteSynchronizationEventListener.java

```diff
@@ -178,22 +273,29 @@ private void submitData(String doc, BaseObject serverConfiguration)
      */
     private void deleteData(String doc, BaseObject serverConfiguration)
     {
-        // FIXME This should be asynchronous; reimplement!
-        HttpPost method = null;
+        HttpPost request = null;
         try {
             String deleteURL = getDeleteURL(serverConfiguration);
             if (StringUtils.isNotBlank(deleteURL)) {
-                this.logger.debug("Pushing deleted document to [{}]", deleteURL);
-                method = new HttpPost(deleteURL);
+                if (this.logger.isDebugEnabled()) {
+                    this.logger.debug("Queueing patient deletion to remote server [{}] (current status: {} active " +
+                            "requests, {} queued requests)",
+                        deleteURL, this.executor.getActiveCount(), this.executor.getQueue().size());
+                }
+                request = new HttpPost(deleteURL);
                 NameValuePair data = new BasicNameValuePair("document", doc);
-                method.setEntity(new UrlEncodedFormEntity(Collections.singletonList(data), Consts.UTF_8));
-                this.client.execute(method).close();
+                request.setEntity(new UrlEncodedFormEntity(Collections.singletonList(data), Consts.UTF_8));
+                request.setConfig(this.requestConfig);
+
+                RemoteSynchronizationRequestThread remoteSynchronizationRequestThread =
+                    new RemoteSynchronizationRequestThread(this.logger, this.client, request);
+                executor.execute(remoteSynchronizationRequestThread);
             }
         } catch (Exception ex) {
-            this.logger.warn("Failed to notify remote server of patient removal: {}", ex.getMessage(), ex);
+            this.logger.warn("Failed to queue patient removal request to remote server: {}", ex.getMessage(), ex);
         } finally {
-            if (method != null) {
-                method.releaseConnection();
+            if (request != null) {
+                request.releaseConnection();
```

Comment:
As discussed, `request.releaseConnection` should probably only happen within the `RemoteSynchronizationRequestThread`, since that is what is responsible for opening the connection and closing it.